dev/breeze/src/airflow_breeze/global_constants.py

```diff
@@ -28,8 +28,10 @@
 from airflow_breeze.utils.host_info_utils import Architecture
 from airflow_breeze.utils.path_utils import AIRFLOW_SOURCES_ROOT, PROVIDER_DEPENDENCIES_JSON_FILE_PATH
 
-RUNS_ON_PUBLIC_RUNNER = "ubuntu-22.04"
-RUNS_ON_SELF_HOSTED_RUNNER = "self-hosted"
+RUNS_ON_PUBLIC_RUNNER = '["ubuntu-22.04"]'
+# we should get more sophisticated logic here in the future, but for now we just check if
+# we use self airflow, vm-based, amd hosted runner as a default
+RUNS_ON_SELF_HOSTED_RUNNER = '["self-hosted", "airflow-runner", "vm-runner", "X64"]'
```

Comment:
I don't seed the labels attached to [the runners](https://github.com/apache/airflow/actions/runners?tab=self-hosted); we need to check before merging to avoid blocking the committers' PRs.

Reply:
I think this should passed as an array instead of a string; it will be serialized to JSON by GHA :thinking:

Reply:
> I don't seed the labels attached to [the runners](https://github.com/apache/airflow/actions/runners?tab=self-hosted); we need to check before merging to avoid blocking the committers' PRs.

Yeah. This one won't get green until that works, so no worries :)

> I think this should passed as an array instead of a string; it will be serialized to JSON by GHA 🤔

It's complicated. We can only print strings as outputs and when needed they need to be converted to arrays with "toJSON" method - but its not obvious how so I am playing with it now to get the right  combination

:shrug: GA woes.

Reply:
Or rather fromJSON :facepalm:

Reply:
You can try:
```
runs-on: ${{ toJSON(steps.selective-checks.outputs.runs-on) }}
```
in the list of outputs

Reply:
your last commit works :tada: 
```
Requested labels: self-hosted, airflow-runner, vm-runner, X64
```

Reply:
Unfortunately not.... Outputs are always Unicode strings - so converting them to JSON just causes subsequent encoding :shrug: ... We need to run `fromJSON` in all the places where the outputs are used as part of workflow definition ...


I think I got it right now... But indeed our runners do not have the right labels defined and do not pick the jobs up:


> Requested labels: self-hosted, airflow-runner, vm-runner, X64
Job defined at: apache/airflow/.github/workflows/ci.yml@refs/pull/35109/merge
Waiting for a runner to pick up this job...